mongo_service.go

```diff
@@ -46,14 +47,41 @@ func (service *MongoService) closeSession() {
 	service.session.Close()
 }
 
-func (service *MongoService) isCurrentNodeMaster() bool {
+func (service *MongoService) isNodeForBackup() bool {
 	db := service.session.DB(defaultDb)
 	result := make(map[string]interface{})
 	err := db.Run(bson.M{"isMaster": 1}, result)
 	if err != nil {
 		log.Panic("Can't check if node is master, db.isMaster() fails", err.Error(), err)
 	}
-	return result["ismaster"].(bool)
+
+	if result["me"] == nil || result["hosts"] == nil {
+		warn.Println("Node is not part of a cluster.")
+		return false
+	}
+
+	master := result["ismaster"].(bool)
+	if master {
+		info.Println("This node is master.")
+		return false
+	}
+
+	thisNode := result["me"].(string)
+	primary := result["primary"].(string)
+	allNodes := result["hosts"].([]string)
+	sort.Strings(allNodes)
+	var lowestSecondary string
+	if allNodes[0] == primary {
+		lowestSecondary = allNodes[1]
+	} else {
+		lowestSecondary = allNodes[0]
+	}
```

Comment:
Currently we have a singleton mongodb-backup service that might run on any MongoDB node. I'm anticipating changing this to 3 instances of the service, one running on each MongoDB node. The one that runs on the MongoDB master has to shut down (as it currently does - so in this event, currently no backup is performed on this run) and all but one of the instances running on secondary nodes should also shut down.

It doesn't have to be the same machine on each run (if, for example, AWS moves machines around and they end up with IPs that order the host names differently, or there's a MongoDB election that results in a new primary node); it's just a simple way of making sure service instances running on different nodes at the same time would order the nodes unambiguously and come to the same conclusion about which instance should continue to perform the backup (so, barring exceptional circumstances where no secondary is available, a backup should always be performed).